main/src/library/Iterator.flix

```diff
@@ -60,6 +60,35 @@ namespace Iterator {
         };
         Iterator(done, next)
 
+    ///
+    /// Returns an iterator that unfolds `f` until the resource at `x` is null.
+    ///
+    /// Assumes that `f` eventually exhausts the resource at `x`.
+    ///
+    /// Assumes `f` may throw a Java Throwable.
+    ///
+    pub def ofThrowable(f: Unit -> a & Impure, x: Ref[a]): Iterator[Result[a, String]] =
```

Comment:
Maybe we should split these concerns into two:

We could add Prelude function called `def try(f: a-> b & Impure): a -> Result[b, String` that lifts an "exceptional function" into a pure function. And then we could call `ofNullable` on that?

Reply:
OK, we should probably not call it `try` but rather `liftTry` or `safeTry` or `Result.try` or whatever.

Reply:
(NB: The function cannot be pure, it has to be effect polymorphism).

Reply:
What's the benefit to adding it to the prelude rather than `Result.try`?